packages/nimble-components/src/table/tests/table-keyboard-navigation.spec.ts

```diff
@@ -0,0 +1,917 @@
+                data.push({
+                    id: i.toString(),
+                    stringData1: `a${i}`,
+                    stringData2: `b${i}`,
+                    stringData3: `c${i}`
+                });
+            }
+
+            await element.setData(data);
+            await connect();
+            await waitForUpdatesAsync();
+        }
+
+        beforeEach(async () => {
+            ({ element, connect, disconnect } = await setupWithTextColumns());
+            pageObject = new TablePageObject<SimpleTableRecord>(element);
+            column1 = element.querySelector<TableColumnText>('#first-column')!;
+            column2 = element.querySelector<TableColumnText>('#second-column')!;
+            column3 = element.querySelector<TableColumnText>('#third-column')!;
+        });
+
+        afterEach(async () => {
+            await disconnect();
+        });
+
+        describe('for a simple table (no hierarchy/grouping/selection) with focus', () => {
+            beforeEach(async () => {
+                await setupBasicTable();
+                element.focus();
+                await waitForUpdatesAsync();
+            });
+
+            it('on initial table focus, the 1st column header is focused if sortable', () => {
+                const firstHeader = pageObject.getHeaderElement(0);
+                expect(currentFocusedElement()).toBe(firstHeader);
+            });
+
+            it('consecutive Enter key presses will sort the focused column (ascending, descending, no sort)', async () => {
+                const firstHeader = pageObject.getHeaderElement(0);
+                expect(firstHeader.sortDirection).toBe(
+                    TableColumnSortDirection.none
+                );
+
+                await sendKeyPress(firstHeader, keyEnter);
+                expect(firstHeader.sortDirection).toBe(
+                    TableColumnSortDirection.ascending
+                );
+
+                await sendKeyPress(firstHeader, keyEnter);
+                expect(firstHeader.sortDirection).toBe(
+                    TableColumnSortDirection.descending
+                );
+
+                await sendKeyPress(firstHeader, keyEnter);
+                await waitForUpdatesAsync();
+                expect(
+                    firstHeader.sortDirection
+                        === TableColumnSortDirection.none
+                        || firstHeader.sortDirection === null
+                ).toBe(true);
+            });
+
+            it('Shift-Enter key presses will sort the focused column (ascending, descending, no sort) and maintain other sorted columns', async () => {
+                const firstHeader = pageObject.getHeaderElement(0);
+                const secondHeader = pageObject.getHeaderElement(1);
+                await sendKeyPress(firstHeader, keyEnter);
+                await sendKeyPressToTable(keyArrowRight);
+
+                await sendKeyPress(secondHeader, keyEnter, { shiftKey: true });
```

Comment:
Similarly, I think our intent is better captured if we send all events to `currentFocusedElement`.

Reply:
Done for these instances. Switching the ones that currently dispatch to the table, to use currentFocusedElement, didn't work though (see other comment).